src/vm.rs

```diff
@@ -489,26 +488,26 @@ impl<'a, S: Storage> Database for VmDb<'a, S> {
     }
 }
 
-pub(crate) struct Vm<'a, S: Storage> {
+pub(crate) struct Vm<'a, S: Storage, C: PevmChain> {
     hasher: &'a ahash::RandomState,
     storage: &'a S,
     mv_memory: &'a MvMemory,
     txs: &'a [TxEnv],
-    chain: Chain,
+    chain: &'a C,
     spec_id: SpecId,
     block_env: BlockEnv,
     beneficiary_location_hash: MemoryLocationHash,
     reward_policy: RewardPolicy,
     new_bytecodes: DeferDrop<DashMap<B256, Bytecode>>,
 }
 
-impl<'a, S: Storage> Vm<'a, S> {
+impl<'a, S: Storage, C: PevmChain> Vm<'a, S, C> {
     pub(crate) fn new(
         hasher: &'a ahash::RandomState,
         storage: &'a S,
         mv_memory: &'a MvMemory,
         txs: &'a [TxEnv],
-        chain: Chain,
+        chain: &'a C,
```

Comment:
Looking back, propagating a lot of PevmChain here seems overkill. We can just pass in a chain_id instead?